### PR TITLE
Add i18n to plugin SEO

### LINKED ISSUE #4543: plugin-seo: Add localization for UI
Currently the UI for this plugin is hard coded in english. We should have translations or at least allow the labels to be set in the config object.

Screenshot 2023-07-20 at 11 42 08 AM

### PR DESCRIPTION
**Pull request by paulpopus**
Closes https://github.com/payloadcms/payload/issues/4543

## Description

Adds the ability to translate the UI of the SEO plugin via the i18n config

With this change the plugin's UI will be translatable via .json files like in core

- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] Existing test suite passes locally with my changes
- [x] I have added new tests with my changes
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/plugin-seo/src/translations.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export default () => {
```
Comment (DanRibbens) [inline packages/plugin-seo/src/translations.ts:1]: I'd like this to be made consistent with how we do translations in Payload.
We have a json file for each language and a schema file that enforces consistency. https://github.com/payloadcms/payload/blob/9da9b1fc5050d4f29bcf6dce2f22027834aaf698/packages/payload/src/translations/translation-schema.json

This way we can move towards having plugins contain all languages without developers needing to add their own translations project-by-project.

File: packages/plugin-seo/src/index.ts [hunk 2]
```diff
@@ -176,6 +178,22 @@ const seo =
 
           return global
         }) || [],
+      i18n: {
+        ...config.i18n,
+        resources: {
+          ...config.i18n?.resources,
+          en: {
```
Comment (DanRibbens) [inline packages/plugin-seo/src/index.ts:185]: I used deepMerge to this in core and will be necessary if we implement my other feedback.

**Review by DanRibbens — CHANGES_REQUESTED** [review]
I'd like to see an e2e test. It looks like we stubbed it out in the repo and didn't add any yet.

File: packages/plugin-seo/src/translations.ts (hunk 1 above)
Comment (paulpopus) [reply packages/plugin-seo/src/translations.ts]: Made the changes as discussed, added e2e tests for the plugin and for the new translations as well

**Review by DanRibbens — APPROVED** [review]

**Pull request merged**